adapter/internal/oasparser/model/policy_container.go

```diff
@@ -0,0 +1,251 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package model
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/ghodss/yaml"
+	"github.com/wso2/product-microgateway/adapter/internal/loggers"
+	"github.com/wso2/product-microgateway/adapter/pkg/logging"
+)
+
+const (
+	policyCCGateway string = "CC"
+)
+
+// policy value validation types
+const (
+	policyValTypeString string = "String"
+	policyValTypeInt    string = "Integer"
+	policyValTypeBool   string = "Boolean" // TODO: check type names with APIM
+	policyValTypeArray  string = "Array"
+	policyValTypeMap    string = "Map"
+)
+
+// PolicyFlow holds list of Policies in a operation (in one flow: In, Out or Fault)
+type PolicyFlow string
+
+const (
+	policyInFlow    PolicyFlow = "request"
+	policyOutFlow   PolicyFlow = "response"
+	policyFaultFlow PolicyFlow = "fault"
+)
+
+// PolicyContainer holds the definition and specification of policy
+type PolicyContainer struct {
+	Specification PolicySpecification
+	Definition    PolicyDefinition
+}
+
+// PolicyContainerMap maps PolicyName -> PolicyContainer
+type PolicyContainerMap map[string]PolicyContainer
+
+// GetFormattedOperationalPolicies returns formatted, Choreo Connect policy from a user templated policy
+func (p PolicyContainerMap) GetFormattedOperationalPolicies(policies OperationPolicies) OperationPolicies {
+	fmtPolicies := OperationPolicies{}
+
+	inFlowStats := policies.In.getStats()
+	for i, policy := range policies.In {
+		if fmtPolicy, err := p.getFormattedPolicyFromTemplated(policy, policyInFlow, inFlowStats, i); err == nil {
+			fmtPolicies.In = append(fmtPolicies.In, fmtPolicy)
+		}
+	}
+
+	outFlowStats := policies.Out.getStats()
+	for i, policy := range policies.Out {
+		if fmtPolicy, err := p.getFormattedPolicyFromTemplated(policy, policyOutFlow, outFlowStats, i); err == nil {
+			fmtPolicies.Out = append(fmtPolicies.In, fmtPolicy)
+		}
+	}
+
+	faultFlowStats := policies.Out.getStats()
+	for i, policy := range policies.Fault {
+		if fmtPolicy, err := p.getFormattedPolicyFromTemplated(policy, policyFaultFlow, faultFlowStats, i); err == nil {
+			fmtPolicies.Fault = append(fmtPolicies.In, fmtPolicy)
```

Comment:
```suggestion
			fmtPolicies.Fault = append(fmtPolicies.Fault, fmtPolicy)
```
Should this change?

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/2656/commits/321b5c37e580d65072fbcd8902e950a87217755a